Daily work summary refactor

surajshetty3416 opened the pull request:
Fixed https://github.com/frappe/erpnext/issues/12376

Now we can have multiple settings for Daily Work Summary which can have different set of users with separate _holiday list_ and _time to send emails_

surajshetty3416 committed:
added daily work summary setting doctype and code to support feature

this will allow multiple setting for daily work summary

surajshetty3416 committed:
added daily work summary setting user doctype

surajshetty3416 committed:
made changes in daily work summary code

surajshetty3416 committed:
[minor] entry change in hr config file

surajshetty3416 committed:
deleted previous daily work summary settings (and its company) doctype

surajshetty3416 committed:
removed unwanted permission check

surajshetty3416 committed:
toggled read_only option for enabled field

surajshetty3416 committed:
removed print statements

surajshetty3416 committed:
add patch for the changes

surajshetty3416 committed:
doc changes

surajshetty3416 committed:
[minor] indentation fix

surajshetty3416 committed:
fixed tests

surajshetty3416 committed:
indentation fixes

surajshetty3416 committed:
codacy issue fix

surajshetty3416 committed:
formatting fixes

netchampfaris commented on erpnext/docs/user/manual/en/human-resources/daily-work-summary.md:
```diff
@@ -1,15 +1,24 @@
 
-You can also choose to Customize the Message you send to your employees:
+**Note:**
+>1. If no holiday list is seleted then the email will be sent every day.
```
selected

surajshetty3416 replied:
https://github.com/surajshetty3416/erpnext/commit/102cd445a34e0c1116ec5576a4fd0592a6297f5f

netchampfaris commented on erpnext/docs/user/manual/en/human-resources/daily-work-summary.md:
```diff
@@ -1,15 +1,24 @@
+**Note:**
+>1. If no holiday list is seleted then the email will be sent every day.
+>2. Setting name will be sent as the title of daily summary email.
+>3. Deactivated settings will be skipped from daily check.
```
What does this mean?

surajshetty3416 replied:
https://github.com/surajshetty3416/erpnext/commit/102cd445a34e0c1116ec5576a4fd0592a6297f5f

olamide2 commented:
Why not some screen shots to allow us appreciate the changes ?

surajshetty3416 committed:
renamed doctype

Renamed Daily Work Summary Setting to  Daily Work Summary Group and did related code and doc changes

surajshetty3416 committed:
fixed typo

surajshetty3416 committed:
updated doc

surajshetty3416 committed:
codacy issue fix

surajshetty3416 committed:
[minor] renamed doctype name in json

surajshetty3416 committed:
Renamed old doctype

surajshetty3416 committed:
fixed indentation

surajshetty3416 committed:
codacy fix

surajshetty3416 committed:
indentation fix

surajshetty3416 committed:
renamed doctype

surajshetty3416 committed:
handled patch exception

surajshetty3416 committed:
fixed exception

rmehta commented on erpnext/hr/doctype/daily_work_summary_group/daily_work_summary_group.py:
```diff
@@ -0,0 +1,59 @@
+	def validate(self):
+		if self.users:
+			if not frappe.flags.in_test and not is_incoming_account_enabled():
+				frappe.throw(_('There must be a default incoming Email Account enabled for this to work. '
```
translatable strings must be in single line. Use smaller strings

Maybe just _('Please enable default incoming account before creating Daily Work Summary Group')

surajshetty3416 replied:
Fixed https://github.com/frappe/erpnext/pull/12944/commits/343192078b2fffb4a4b524731713d422e7f9dd89

surajshetty3416 committed:
Update daily_work_summary_group.py

surajshetty3416 committed:
rename patch file

removed abbreviation in file name

surajshetty3416 committed:
handled exception in patch code

surajshetty3416 committed:
removed Unnecessary pass statement

surajshetty3416 committed:
[minor] indentation fix
